client/signup/steps/plans/util.ts

```diff
@@ -6,9 +6,16 @@ type DomainItem = {
 	product_slug?: string;
 };
 
-export const getIntervalType = (): string => {
-	const urlParts = getUrlParts( typeof window !== 'undefined' ? window.location?.href : '' );
-	const intervalType = urlParts?.searchParams.get( 'intervalType' ) || '';
+export const getIntervalType = ( path?: string ): string => {
+	const pathHasIntervalType = path?.includes( 'intervalType=' );
+
+	if ( pathHasIntervalType ) {
+		const urlPartsFromPath = getUrlParts( path || '' );
```

Comment:
It feels a little over-defensive to have `path || ''`.

We're checking if the path includes `intervalType=` by string matching, and if does, then `path` already exists and is not empty. Here's how I'd write this function:

```js
export const getIntervalType = ( path?: string ): string => {
  const url = path ?? window?.location?.href ?? '';
  const intervalType = getUrlParts( url ).searchParams.get( 'intervalType' ) ?? 'yearly';

  ... /* keep the rest as it is */
}
```